MDEV-21331: sync preinst and postrm script

fauust opened the pull request:
- [x] *The Jira issue number for this PR is: [MDEV-21331](https://jira.mariadb.org/browse/MDEV-21331)*

Not exactly related to but the same problem could appear on uninstallation.

## Description

When detecting running mysqld/mariadbd processes to stop, ignore processes in different namespaces.

## How can this PR be tested?

Install mariadb and start a mariadb docker container on the same host, test upgrade/uninstall.

ottok commented:
This change was originally made by @vuvova on 10.2 in https://github.com/MariaDB/server/commit/fb7c1b9415c9a8b0dc2e86ae44f0e7a2634e5d7e but he omitted the prerm script:

```
± grep -F 'pgrep' debian/*
debian/mariadb-server-10.6.postrm:    if ! pgrep -x mariadbd > /dev/null; then return; fi
debian/mariadb-server-10.6.postrm:    if ! pgrep -x mysqld > /dev/null; then return; fi
debian/mariadb-server-10.6.preinst:    if ! pgrep -x --ns $$ mariadbd > /dev/null; then return; fi
debian/mariadb-server-10.6.preinst:    if ! pgrep -x --ns $$ mysqld > /dev/null; then return; fi
```

Change is OK but should probably go on oldest possible branch and not on 10.5?

fauust commented:
> This change was originally made by @vuvova on 10.2 in [fb7c1b9](https://github.com/MariaDB/server/commit/fb7c1b9415c9a8b0dc2e86ae44f0e7a2634e5d7e) but he omitted the postrm script:

And the `-x` is missing too.

> Change is OK but should probably go on oldest possible branch and not on 10.5?
If https://github.com/MariaDB/server/pull/2068 is merged in 10.2, it should also hit 10.3 and 10.4.

pgrep was introduced to postrm script in  10.5:

```console
❯ git checkout 10.4
Updating files: 100% (5109/5109), done.
Switched to branch '10.4'
Your branch is up to date with 'origin/10.4'.

❯ git grep pgrep
mariadb-server-10.4.preinst:    if ! pgrep -x --ns $$ mysqld > /dev/null; then return; fi

❯ git checkout 10.5
Updating files: 100% (5109/5109), done.
Switched to branch '10.5'
Your branch is up to date with 'upstream/10.5'.

❯ git grep pgrep
mariadb-server-10.5.postrm:    if ! pgrep -x mariadbd > /dev/null; then return; fi
mariadb-server-10.5.postrm:    if ! pgrep -x mysqld > /dev/null; then return; fi
mariadb-server-10.5.preinst:    if ! pgrep -x --ns $$ mariadbd > /dev/null; then return; fi
mariadb-server-10.5.preinst:    if ! pgrep -x --ns $$ mysqld > /dev/null; then return; fi
```

ottok reviewed (commented):
Once https://github.com/MariaDB/server/pull/2076 is merged across all branches all the way up to 10.9 it will make this PR obsolete.

an3l commented:
Closing this PR, pushed with https://github.com/MariaDB/server/commit/794bebf9ee18de4138c4d2c93853d49ff3af7f12
